File: TowerOfHanoi.py
class TOH(object):
    def __init__(self, N):
        self.N = N
        self.source = "Source"
        self.helper = "Helper"
        self.destination = "Destination"

    def _towerOfHanoi(self):
        self.__hypothesis(self.N, self.source, self.helper, self.destination)

    def __hypothesis(self, numOfPlates, src, helper, dest):
        if numOfPlates == 1:
            # Base condition
            print(f"Moving plate {numOfPlates} from {src} to {dest}")
            return 

        self.__hypothesis(numOfPlates-1, src, dest, helper)
        # Induction
        print(f"Moving plate {numOfPlates} from {src} to {dest}")
        self.__hypothesis(numOfPlates-1, helper, src, dest)
        
numberOfPlates = 5
obj = TOH(numberOfPlates)
obj._towerOfHanoi()

File: Java/TowerOfHanoi.java
package Java;
/**
 * @author Pallavi
 */
public class TowerOfHanoi
{
    /**
     * 
     * @param N - number of disks
     * @param source - source tower
     * @param destination - destination tower
     * @param helper - intermediate tower
     */

    public static void towerOfHanoi(int N, int source, int destination, int helper)
    {

    if(N==1)
    {
        System.out.println("Moving plate " + N + " from " + source +" to " + destination);
        return;

    }

    towerOfHanoi(N - 1, source, helper, destination);
    System.out.println("Moving plate " + N + " from " + source +" to " + destination);
    towerOfHanoi(N-1, helper, destination, source);

    }

/**
 * Main function 
 * 
 */
    public static void main(String[] args)
    {
    int N = 5;
    int source = 1;
    int helper = 2;
    int destination = 3;

    towerOfHanoi(N, source, destination, helper);

    }

    
}